Training5/src/task-1/src/App.tsx

```diff
@@ -0,0 +1,97 @@
+import React, {useEffect, useState} from 'react';
+import './App.css';
+import Todo from "./component/todo";
+import axios from 'axios';
+import {useDispatch, useSelector} from "react-redux";
+import {ITodo} from "./interfaces/todo.interface";
+import AddTodoForm from "./component/add-todo-form";
+import {getTodoList, getTodoListByFilter} from "./reducer/todo.slice";
+import SearchFilterTodo from "./component/search-filter-todo";
+
+const base_url = 'http://localhost:9000/todos'
+
+function App() {
+    const [loading, setLoading] = useState(false);
+
+    useEffect(() => {
+        fetchTodoList()
+    }, [])
+
+    const dispatch = useDispatch();
+    const todoStore = useSelector((state: any) => state.todos);
+
+    const fetchTodoList = () => {
+        setLoading(true)
+        axios.get(`${base_url}`).then((res) => {
+            setLoading(false);
+            dispatch(getTodoList(res.data))
+        })
+    }
+
+    const renderTodoList = (data: ITodo[]) => {
+        return data.map((data) => {
+            return <Todo key={data.id} id={data.id} name={data.name} completed={data.completed}
+                         handleDelete={handleDelete}
+                         handleCompleteOrIncomplete={handleCompleteOrIncomplete}/>
+        })
+    }
+
+    const handleDelete = (id: string) => {
+        axios.delete(`${base_url}/${id}`).then(() => {
+            fetchTodoList()
+        }).catch()
+    }
+
+    const handleCompleteOrIncomplete = (id: string, todoStatus: boolean) => {
+        axios.put(`${base_url}/${id}`, {
+            completed: !todoStatus
+        }).then(() => {
+            fetchTodoList()
+        }).catch()
+    }
+
+    const handleAddTodo = (name: string) => {
+        axios.post(`${base_url}`, {
+            name: name,
+            completed: false,
+        }).then(() => {
+            fetchTodoList()
+        }).catch()
+    }
+
+    const handleSearchFilter = ({name, status}: { name: string, status: boolean }) => {
```

Comment:
we should not treat it as a new action, we should implement it just using a selector + useMemo, the data source (to-do list) should not be changed just depend on the search or filter value